### PR TITLE
Add channel's intended usage for deposit/withdrawal

### PR DESCRIPTION
**Pull request by velzevur**
This is a squashed version of #126

### CODE REVIEW

File: epoch/api/channels_api_usage.md [hunk 1]
```diff
@@ -518,3 +521,268 @@ details:
...
+closing a channel on-chain, the closing balances of the participants are
+checked against this balance. Under no circumstances can the sum of the closing balances
+be greater than the total balance on-chain.
+
+Participants are able to modify the total balance: the following two functionalities are available:
+
+* deposit - when a participant wants to commit more tokens from his on-chain
+  balance to the channel total balance
+* withdrawal - when a participant wants to transfer tokens out of the channel
+  on-chain balace to one's personal account
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: balace -> balance
Should we mention that a deposit/withdrawal of 0 could work as a mutually agreed snapshot operation?

File: epoch/api/channels_api_usage.md [hunk 2]
```diff
@@ -518,3 +521,268 @@ details:
...
+  | state_hash | string | the root of the internal channel state hash after the withdraw |
+  | round | integer | the next channel round |
+  | nonce | integer | withdrawer's nonce |
+
+#### Start withdraw
+
+Any of the participants can initiate a withdrawal. The only requirements are:
+* Channel is already opened
+* No off-chain update/deposit/withdrawal is currently being performed
+* Channel is not being closed or in a solo closing state
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: * The withdrawal amount must be equal to or greater than zero, and cannot exceed the available balance on the channel (minus the `channel_reserve`).

File: epoch/api/channels_api_usage.md [hunk 3]
```diff
@@ -518,3 +521,268 @@ details:
...
+  | state_hash | string | the root of the internal channel state hash after the deposit |
+  | round | integer | the next channel round |
+  | nonce | integer | depositor's nonce |
+
+#### Start deposit
+
+Any of the participants can initiate a deposit. Only requirements are:
+* Channel is already opened
+* No off-chain update/deposit/withdrawal is currently being performed
+* Channel is not being closed or in a solo closing state
```
Comment (uwiger) [inline epoch/api/channels_api_usage.md]: * The deposit must be equal to or greater than zero, and cannot exceed the available balance on the depositor's account.

File: epoch/api/channels_api_usage.md (hunk 1 above)
Comment (velzevur) [reply epoch/api/channels_api_usage.md]: I plan on adding this when describing the solo-snapshot tx

**Pull request merged**